src/hmem_ze.c

```diff
@@ -330,20 +331,20 @@ static int ze_hmem_init_fds(void)
 			continue;
 
 		memset(dev_name, 0, sizeof(dev_name));
-		strncpy(dev_name, dev_dir, sizeof(dev_name));
-		strncat(dev_name, "/",
-			sizeof(dev_name) - strlen(dev_name));
-		strncat(dev_name, ent->d_name,
-			sizeof(dev_name) - strlen(dev_name));
+		if (snprintf(dev_name, NAME_MAX, "%s%s", dev_dir, ent->d_name) < 0)
```

Comment:
This checks for an error, but will not detect truncation.

Reply:
Ahh I assumed truncation was assumed to be an error. Updated.